### PR TITLE
Bugfix/use default branch

### LINKED ISSUE #55: Daemon raises an error when no 'master' is detected
## Problem
When there is no `master` branch on the repository, daemon raises an error `Branch Not Found`:
```
WARN: Exception raised by code in GitPoller#poll
  WARN: Waiting 10s and restarting the daemon
  WARN: Server responded with code 404, message: 404 Branch Not Found. Request URI: https://${....}
  WARN:   /home/ci/.local/share/autoproj/gems/ruby/2.7.0/bundler/gems/autoproj-daemon-6c5c9671fd8b/lib/autoproj/daemon/git_api/services/gitlab.rb:151:in `rescue in exception_adapter'
  WARN:   /home/ci/.local/share/autoproj/gems/ruby/2.7.0/bundler/gems/autoproj-daemon-6c5c9671fd8b/lib/autoproj/daemon/git_api/services/gitlab.rb:148:in `exception_adapter'
  WARN:   /home/ci/.local/share/autoproj/gems/ruby/2.7.0/bundler/gems/autoproj-daemon-6c5c9671fd8b/lib/autoproj/daemon/git_api/services/gitlab.rb:116:in `delete_branch'
```

## Possible solution
Use `try_resolve` from autoproj git importer

### PR DESCRIPTION
**Pull request by ivan1993br**
## Feature
- Uses `try_resolve_remote_head_from_server` from `Autobuild::Git` to resolve `default_branch`
- Close #55 

## Some Points
- I use ~`try_resolve_remote_head_from_server` directly from the `Autobuild::Git` which will directly requires this API from this importer~`resolve_remote_head`, for me, it made sense to use it, we could re-implement a solution here if we do not want to use the importer API

- I made a "copy-paste" on the test `untar`, `no-master` stuff from `Autobuild`, I do not know if it was the best idea, it was the fattest and more reliable that I saw, but you guys see what you think, if I should put in the test helper, use another approach, etc....

Take a look and see what it needs to be improved, I am open to any change,

Thanks!

### CODE REVIEW

File: lib/autoproj/daemon/package_repository.rb [hunk 1]
```diff
@@ -46,7 +47,13 @@ def repo_url
 
             # @return [String]
             def branch
-                vcs[:remote_branch] || vcs[:branch] || "master"
+                current_importer = autobuild.importer
+
+                vcs[:remote_branch] ||
+                    vcs[:branch] ||
+                    current_importer.try_resolve_remote_head_from_server(
```
Comment (g-arjones) [inline lib/autoproj/daemon/package_repository.rb]: 1. Use `Autobuild::Git#resolve_remote_head` instead.
2. Don't create the importer unless you have to (if both `vcs[:remote_branch]` and `vcs[:branch]` are unset)

File: lib/autoproj/daemon/package_repository.rb [hunk 2]
```diff
@@ -46,7 +47,14 @@ def repo_url
 
             # @return [String]
             def branch
-                vcs[:remote_branch] || vcs[:branch] || "master"
+                curr_branch = vcs[:remote_branch] || vcs[:branch]
+                return curr_branch if curr_branch
+
+                current_importer = autobuild.importer
+
+                current_importer.resolve_remote_head(
```
Comment (doudou) [inline lib/autoproj/daemon/package_repository.rb]: Just `autobuild.importer.resolve_remote_head`

File: lib/autoproj/daemon/package_repository.rb [hunk 3]
```diff
@@ -46,7 +47,14 @@ def repo_url
 
             # @return [String]
             def branch
-                vcs[:remote_branch] || vcs[:branch] || "master"
+                curr_branch = vcs[:remote_branch] || vcs[:branch]
```
Comment (doudou) [inline lib/autoproj/daemon/package_repository.rb]: `explicit_branch` ?

**Review by doudou — APPROVED** [review]
Just small style/minor fixes.

**Comment by g-arjones** [comment]
Looks good to me as well. Please, make sure you ran the whole test suite and make sure it passes since you are changing a critical code path.

**Comment by ivan1993br** [comment]
Yeah... I have just saw that there is one broken test, I am fixing it now

**Comment by ivan1993br** [comment]
Can you guys approve the `workflow run`?

**Comment by g-arjones** [comment]
Thanks!

**Review by g-arjones — APPROVED** [review]

**Pull request merged**